--- /usr/lib/rpm/redhat/brp-java-repack-jars.orig	2009-04-20 09:56:37.000000000 -0600
+++ /usr/lib/rpm/redhat/brp-java-repack-jars	2009-04-20 10:01:14.000000000 -0600
@@ -38,8 +38,8 @@
 		rm -f $j
 		
 		# Create the directories first.
-		for d in `find -type d | LC_ALL=C sort`; do
-            		mkdir -p $JARDIR/$d
+		find -type d | LC_ALL=C sort | while read d; do
+            		mkdir -p "$JARDIR/$d"
         	done 
 
 		# Get the modtime from the newest ChangeLog. If the project
@@ -64,14 +64,14 @@
 
 		# move the contents over to the a new directory in order and set
 		# the times. 
-		for f in `find -type f | LC_ALL=C sort`; do 
-            		cp $f $JARDIR/$f
-            		touch --date="$DATE" $JARDIR/$f
+		find -type f | LC_ALL=C sort | while read f; do 
+            		cp "$f" "$JARDIR/$f"
+            		touch --date="$DATE" "$JARDIR/$f"
         	done
 		popd > /dev/null
 
 		# Set the times of the directories.
-		touch --date="$DATE" `find $JARDIR -type d`
+		find $JARDIR -type d -print0 | xargs -0 touch --date="$DATE"
 
 		# make the jar
 		pushd $JARDIR > /dev/null
