diff -urNp redhat-rpm-config-9.0.3_old/kmodtool redhat-rpm-config-9.0.3/kmodtool
--- redhat-rpm-config-9.0.3_old/kmodtool	2010-08-09 16:36:43.581511561 -0400
+++ redhat-rpm-config-9.0.3/kmodtool	2010-08-09 17:05:50.600326999 -0400
@@ -172,8 +172,7 @@ EOF
 
 cat <<EOF
 %preun         -n kmod-${kmod_name}${dashvariant}
-rpm -ql kmod-${kmod_name}${dashvariant} | grep '\.ko$' \
-    > /var/run/rpm-kmod-${kmod_name}${dashvariant}-modules
+rpm -ql kmod-${kmod_name}${dashvariant}-%{kmod_version}-%{kmod_release}.$(arch) | grep '\.ko$' > /var/run/rpm-kmod-${kmod_name}${dashvariant}-modules
 EOF
         
 cat <<EOF
